src/CMakeLists.txt

```diff
@@ -25,24 +28,60 @@ endif()
 add_definitions(-DSRC_DIR="${CMAKE_SOURCE_DIR}")
 add_definitions(-DTESTCASE_DIR="${CMAKE_SOURCE_DIR}/../testcase")
 
+
 add_subdirectory(third_party/gtest-1.7.0)
 include_directories(${gtest_SOURCE_DIR}/include ${gtest_SOURCE_DIR})
 include_directories(../include)
 
+mark_as_advanced(
+gtest_build_samples
+gtest_build_tests
+gtest_disable_pthreads
+gtest_force_shared_crt
+)
+
+set_target_properties(gtest gtest_main 
+    PROPERTIES FOLDER "Third_party")
+
+if(MSVC)
+    if (MSVC_VERSION GREATER_EQUAL 1900)
+        target_compile_definitions(gtest PUBLIC _SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING)
+        target_compile_definitions(gtest_main PUBLIC _SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING)
+    endif()
+endif()
+
 add_executable(parse_stdin parse_stdin.cc)
 add_executable(parse_file parse_file.cc)
 add_executable(parse_file2 parse_file_2.cc)
 
+if(MSVC)
+	target_compile_definitions(parse_stdin PUBLIC _CRT_SECURE_NO_WARNINGS)
+	target_compile_definitions(parse_file PUBLIC _CRT_SECURE_NO_WARNINGS)
+	target_compile_definitions(parse_file2 PUBLIC _CRT_SECURE_NO_WARNINGS)
+endif()
+
 function(add_toml_test target)
-    add_executable(${target}_test ${target}_test.cc)
+    add_executable(${target}_test ${target}_test.cc ../include/toml/toml.h)
```

Comment:
Adding .h seems weird to me. Here, adding toml.h as an input src. It will work, though...

Reply:
maybe you just want to add toml.h as a dependency, right?
Then what we need is target_include_directories (or include_directories).
https://stackoverflow.com/questions/13703647/how-to-properly-add-include-directories-with-cmake

Reply:
The issue is not about code dependencies, I think that was working correctly as it was.  It was about actually seeing the project in visual studio.  And the basic issue was toml.h wasn't showing up anywhere in the project hierarchy, so the project was kind of a pain to work with as the main file wasn't showing up anywhere.    So this isn't adding much if anything to the chain,  but it does make it lot easier to work with the project and target in Visual studio, and shouldn't affect it in other builds.  

The other option is to make a dummy target with just toml.h in it to work with, but this is what I have seen done in some other header only libraries and have used in the past.

Reply:
Hmm, I haven't cared for *.vcproj at al... I'm not so sure what the best practice is when considering all platforms, but ok for now.